bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -1087,7 +1089,7 @@ protected synchronized RuleStatusInfo getRuleStatusInfo(String rUID) {
         return statusMap.get(rUID);
     }
 
-    protected synchronized String getUniqueId() {
+    protected synchronized @Nonnull String getUniqueId() {
```

Comment:
I'll not add any null checks because I know that it does not return null. The problem with the annotation is it adds unnecessary checks on each method call.